Automattic/wp-calypso
# Domain Upsell Back Button: Previous Page or Preferred View

7639-gh-Automattic/dotcom-forge

## Proposed Changes

Have the "Home" link on the domain with plan upsell page take you to either wp-admin or My Home, depending on your current site admin interface preferences. If this value isn't available, we'll keep using the most recent page from browser history as a backup.

The issue linked above recommends using browser history first and My Home/wp-admin as the fallback, but we've shifted away from that for the reasons [discussed in the comments](https://github.com/Automattic/wp-calypso/pull/91957#discussion_r1654574140).

## Why are these changes being made?


Sending a site owner who prefers wp-admin to My Home could be jarring, and not the best experience, since they've expressed that this isn't how the prefer to interact with their site.

## Testing Instructions

1. Open WordPress.com and select a site free site with no plan or domain name.
2. Make note of your test site's ID
3. Simple sites don't currently have access to the site admin interface setting, but there is work underway (see: pfsHM7-EF-p2) to change that. You can activate the upcoming setting on your test site by ssh'ing into your sandbox and running:
	- `wpsh`
	- `update_blog_option( blog_id, 'wpcom_admin_interface', 'wp-admin' );`
	- `update_blog_option( blog_id, 'wpcom_classic_early_release', 1 );`
4. Locate the "Free domain with annual plan" nudge and click the "upgrade" button
5. Confirm that the "Home" button takes you to your test site's wp-admin (note: this assumes you ran the above commands to change your site's interface preference to wp-admin).
6. To switch back to the default/calypso interface, run the following command via `wpsh` on your sandbox:
	- `update_blog_option( blog_id, 'wpcom_admin_interface', 'default' );`
7. Visit your test site's My Home page locally at `calypso.localhost:3000/home/[SITE_URL]. Make sure you manually get yourself back to your local Calypso, because that wp-admin link in the previous step will have taken you to production!
8. Once you've loaded your My Home page, test the Upgrade button and the Home link again
9. Confirm that this time, with the interface setting changed, the button takes you back to My Home, and not wp-admin

<details>
<summary>10. To simulate a scenario where our desired "home" link isn't available, apply the following diff:</summary>

```diff
diff --git a/client/my-sites/domains/domain-search/index.jsx b/client/my-sites/domains/domain-search/index.jsx
index 12b422f4ab..b72ea4dfb6 100644
--- a/client/my-sites/domains/domain-search/index.jsx
+++ b/client/my-sites/domains/domain-search/index.jsx
@@ -319,10 +319,7 @@ class DomainSearch extends Component {
 							goBackLink: `/setup/${ siteIntent }/launchpad?siteSlug=${ selectedSiteSlug }`,
 					  }
 					: {
-							goBackLink:
-								this.props.preferredView === 'wp-admin'
-									? this.props.wpAdminUrl
-									: `/home/${ selectedSiteSlug }`,
+							goBackLink: '',
 					  };
 			content = (
 				<span>

```

</details>

11. Navigate to a different area of Calypso (ie, Pages, or Posts) and test the Upgrade button and Home links again. Confirm that they return you to your most recent page.
